lib/components/Atom/WheelPicker/index.tsx

```diff
@@ -0,0 +1,231 @@
+import {
+    HTMLAttributes,
+    KeyboardEvent,
+    useEffect,
+    useMemo,
+    useRef,
+    useState,
+} from "react";
+import { DropdownItem } from "../dropdown";
+import clsx from "clsx";
+import "./wheel-picker.scss";
+import { InputProps } from "@components/Input/base";
+
+export interface WheelPickerProps extends HTMLAttributes<HTMLElement> {
+    data: { value: string | number }[];
+    selectedValue: string | number;
+    setSelectedValue: (value: string | number) => void;
+    handleKeyDown: (e: React.KeyboardEvent) => void;
+    focus: boolean;
+    dropDownItemClassName?: string;
+    containerClassName?: string;
+    listClassName?: string;
+}
+
+export const WheelPicker = ({
+    data,
+    selectedValue,
+    setSelectedValue,
+    handleKeyDown,
+    focus,
+    dropDownItemClassName,
+    containerClassName,
+    listClassName,
+    ...rest
+}: WheelPickerProps) => {
+    const dataRefs = useRef<HTMLDivElement[]>([]);
+    const dataItemsMap = useMemo(
+        () =>
+            data.reduce(
+                (map, item, index) => map.set(item.value, index),
+                new Map(),
+            ),
+        [data],
+    );
+
+    const currentDataValue = useRef(dataItemsMap.get(selectedValue) ?? 0);
+    const dataItemsContRef = useRef<HTMLUListElement>(null);
+
+    const handleDataScroll = (event: Event) => {
+        let isAnimating = false;
+        if (!isAnimating) {
```

Comment:
Sorry, didn't get: we are assign a value (`false`) to a variable (`isAnimating`) and then checking it?

Reply:
I will remove this , had forgotten to remove it . I was trying to put this code as callback for requestAnimationFrame, so had added it